vermouth/gmx/itp_read.py

```diff
@@ -0,0 +1,533 @@
+            effector_param_str = effector_param_str[:-1]  # Remove the closing parenthesis
+            try:
+                effector_class = PARAMETER_EFFECTORS[effector_name]
+            except KeyError:
+                raise IOError('{} is not a known parameter effector.'
+                              .format(effector_name))
+            if '|' in effector_param_str:
+                effector_param_str, effector_format = effector_param_str.split('|')
+            else:
+                effector_format = None
+            effector_param = [elem.strip() for elem in effector_param_str.split(',')]
+            parameter = effector_class(effector_param, format_spec=effector_format)
+        else:
+            parameter = token
+        parameters.append(parameter)
+    return parameters
+
+
+def _is_param_effector(token):
+    return (
+        '(' in token
+        and not token.startswith('(')
+        and token.endswith(')')
+    )
+
+
+def _base_parser(tokens, context, context_type, section, current_meta, natoms=None, delete=False):
+
+    # Group the atoms and their attributes
+    atoms = _get_atoms(tokens, natoms)
+
+    if natoms is not None and len(atoms) != natoms:
+        raise IOError('Found {} atoms while {} were expected.'
+                      .format(len(atoms), natoms))
+
+    # Normalize the atom references.
+    # Blocks and links treat these references differently.
+    # For blocks:
+    # * references can be written as indices or atom names
+    # * a reference cannot be prefixed by + or -
+    # * an interaction cannot create a new atom
+    # For links:
+    # * references must be atom names, but they can be prefixed with one or
+    #   more + or - to signify the order in the sequence
+    # * interactions create nodes
+
+    treated_atoms = _treat_block_interaction_atoms(atoms, context, section)
+
+    # Everything that is not atoms are the interaction parameters    
+    parameters = _parse_interaction_parameters(tokens)
+
+    apply_to_all_interactions = context._apply_to_all_interactions[section]
+
+    if current_meta:
+       meta = {current_meta['condition']:current_meta['tag']}
+    else:
+       meta = {} #dict(collections.ChainMap(meta, apply_to_all_interactions))
+ 
+    interaction = Interaction(
+                  atoms=treated_atoms,
+                  parameters=parameters,
+                  meta=meta,
+                  )
+
+    interaction_list = context.interactions.get(section, [])
+    interaction_list.append(interaction)
+    context.interactions[section] = interaction_list
+
+
+def _parse_block_atom(tokens, context):
+
+    # deque does not support slicing
+    first_six = (tokens.popleft() for _ in range(6))
+    index, atype, resid, resname, name, charge_group = first_six
+    
+    if str(index) in context:
+        msg = ('There is already an atom named "{}" in the block "{}". '
+               'Atom names must be unique within a block.')
```

Comment:
error message no longer consistent with the actual check

Reply:
done